packages/ui/src/views/repo/pull-request/diff-viewer/pull-request-diff-viewer.tsx

```diff
@@ -0,0 +1,321 @@
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react'
+
+import { Card } from '@components/card'
+import { Input } from '@components/input'
+import { Text } from '@components/text'
+import { DiffFile, DiffModeEnum, DiffView, DiffViewProps, SplitSide } from '@git-diff-view/react'
+import { DiffBlock } from 'diff2html/lib/types'
+import { debounce } from 'lodash-es'
+import { OverlayScrollbars } from 'overlayscrollbars'
+
+import constants from './constants'
+
+const TextArea = ({ onChange }: { onChange: (v: string) => void }) => {
+  const [val, setVal] = useState('')
+
+  useEffect(() => {
+    onChange(val)
+  }, [val])
+
+  return (
+    <textarea
+      className="min-h-[80px] w-full border !p-[2px]"
+      autoFocus
+      value={val}
+      onChange={e => setVal(e.target.value)}
```

Comment:
Can we directly call onChange here along with setVal instead of using a useEffect?

Reply:
done